### PR TITLE
Rewrite Ballerina Library Release Pipeline with Ballerina SwanLake

### PR DESCRIPTION
**Pull request by ThisaruGuruge**
## Purpose

Fixes: #6457 

This will;

- Migrate the release script to Ballerina SwanLake
- Improve the dependency graph generation to include the Ballerina extended modules
- Fix some known issues in the dashboard generation and release workflows

### CODE REVIEW

File: dashboard/gitihub.bal [hunk 1]
```diff
@@ -0,0 +1,178 @@
+// Copyright (c) 2022, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```
Comment (niveathika) [inline dashboard/gitihub.bal:1]:
```suggestion
// Copyright (c) 2024, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```

**Review by niveathika — APPROVED** [review]
LGTM other than a minor comment

**Review by daneshk — DISMISSED** [review]

**Comment by ThisaruGuruge** [comment]
Waiting for the GitHub connector fix #6456

File: dashboard/github.bal [hunk 2]
```diff
@@ -0,0 +1,185 @@
+// Copyright (c) 2022, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+//
+// WSO2 Inc. licenses this file to you under the Apache License,
```
Comment (NipunaRanasinghe) [inline dashboard/github.bal:3]:
```suggestion
// Copyright (c) 2024, WSO2 LLC. (http://www.wso2.org)
//
// WSO2 LLC. licenses this file to you under the Apache License,
```

File: dashboard/github.bal [hunk 3]
```diff
@@ -0,0 +1,185 @@
...
+// under the License.
+
+import ballerina/io;
+import ballerina/lang.regexp;
+import ballerina/mime;
+import ballerina/os;
+import ballerina/url;
+import ballerinax/github;
+
+int issueCount = 0;
```
Comment (NipunaRanasinghe) [inline dashboard/github.bal:24]: Is this used anymore?

File: dashboard/github.bal [hunk 4]
```diff
@@ -0,0 +1,185 @@
...
+        interval: HTTP_REQUEST_DELAY_IN_SECONDS,
+        backOffFactor: HTTP_REQUEST_DELAY_MULTIPLIER
+    },
+    auth: {token}
+});
+
+function getDefaultBranch(string moduleName) returns string|error {
+    github:FullRepository repository = check github->/repos/[GITHUB_ORG]/[moduleName];
+    return repository.default_branch;
+
```
Comment (NipunaRanasinghe) [inline dashboard/github.bal:43]: Additional blank line?

File: dashboard/github.bal [hunk 5]
```diff
@@ -0,0 +1,185 @@
...
+        };
+    }
+    string moduleName = module.name;
+    string defaultBranch = module.default_branch;
+    return {
+        name: "CodeCov",
+        badgeUrl: string `${CODECOV_BADGE_URL}/${BALLERINA_ORG_NAME}/${moduleName}/branch/${defaultBranch}/graph/badge.svg`,
+        htmlUrl: string `${CODECOV_BADGE_URL}/${BALLERINA_ORG_NAME}/${moduleName}`
+    };
+
```
Comment (NipunaRanasinghe) [inline dashboard/github.bal:154]: Additional blank line?

**Review by NipunaRanasinghe — COMMENTED** [review]
@ThisaruGuruge Any reason why we are not commiting the `dependencies.toml` files for the Ballerina packages (`dashboard` and `release`) we have in here.  Wouldn't it be a better option to keep the pipeline fail-proof against any breaking changes coming from the dependencies (e.g. the github connector) in the future? Sorry If I'm missing something  here.

**Review by NipunaRanasinghe — APPROVED** [review]
LGTM

**Pull request merged**